File: src/main/java/com/matag/game/turn/action/permanent/GainControlPermanentService.java
package com.matag.game.turn.action.permanent;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

import com.matag.game.cardinstance.CardInstance;

@Component
public class GainControlPermanentService {
  private static final Logger LOGGER = LoggerFactory.getLogger(GainControlPermanentService.class);

  public void gainControlUntilEndOfTurn(CardInstance target, String newControllerName) {
    target.getModifiers().getModifiersUntilEndOfTurn().setNewController(newControllerName);
    LOGGER.info("Changed controller until end of turn of " + target.getIdAndName() + " to " + newControllerName);
  }
}

File: src/main/kotlin/com/matag/game/turn/action/permanent/GainControlPermanentService.kt
package com.matag.game.turn.action.permanent

import com.matag.game.cardinstance.CardInstance
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

@Component
class GainControlPermanentService {
    fun gainControlUntilEndOfTurn(target: CardInstance, newControllerName: String?) {
        target.modifiers.modifiersUntilEndOfTurn.newController = newControllerName
        LOGGER.info("Changed controller until end of turn of " + target.idAndName + " to " + newControllerName)
    }

    companion object {
        private val LOGGER: Logger = LoggerFactory.getLogger(GainControlPermanentService::class.java)
    }
}
